python/tvm/relay/backend/graph_runtime_codegen.py

```diff
@@ -348,12 +348,37 @@ def _get_json(self):
             attrs["device_index"] = ["list_int", device_types]
         attrs["dltype"] = ["list_str", dltypes]
 
+        # Metadata definitions
+        graph_inputs = {}
+        graph_outputs = {}
+        for node_id in arg_nodes:
+            node_name = nodes[node_id]['name']
+            if node_name not in self.params:
+                graph_inputs[node_name] = {}
+                graph_inputs[node_name]['id'] = node_id
+                graph_inputs[node_name]['dtype'] = dltypes[node_id]
+                graph_inputs[node_name]['shape'] = shapes[node_id]
+        for node_id in heads:
+            node_name = nodes[node_id[0]]['name']
+            graph_outputs[node_name] = {}
+            graph_outputs[node_name]['id'] = node_id[0]
+            graph_outputs[node_name]['dtype'] = dltypes[node_id[0]]
+            graph_outputs[node_name]['shape'] = shapes[node_id[0]]
+
+        metadata = {}
+        metadata['signatures'] = {}
+        metadata['signatures']['default'] = {}
+        metadata['signatures']['default']['inputs'] = graph_inputs
+        metadata['signatures']['default']['outputs'] = graph_outputs
+
+        # Keep  'metadata' always at end
         json_dict = {
```

Comment:
Don't you need OrderedDict or similar here? Python dicts and json.dumps don't guarantee order, see https://stackoverflow.com/questions/10844064/items-in-json-object-are-out-of-order-using-json-dumps.  It does look a little scary having this implicit ordering, can't graph_runtime just skip the metadata key instead of stopping parsing?

Reply:
I think JSONParser doesn't support complete skip of an object now. I will use the OrderedDict for now.

Reply:
The dict semantics in new version of Pythons has changed. There are a few places where we have been writing code with them in mind, we should probably be careful. I've fixed a couple bugs in the past related to this assumption.